File: tests/rosettacode/binsearch.py
# Binary search - find 7 in sorted array
def binary_search(arr, target):
    lo, hi = 0, len(arr) - 1
    while lo <= hi:
        mid = (lo + hi) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            lo = mid + 1
        else:
            hi = mid - 1
    return -1

arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
target = 7
idx = binary_search(arr, target)
if idx >= 0:
    print(f"Found {target} at index {idx}")
else:
    print(f"{target} not found")

File: tests/rosettacode/binsearch.clj
;; Binary search - find 7 in sorted array
(defn binary-search [arr target]
  (loop [lo 0 hi (dec (count arr))]
    (when (<= lo hi)
      (let [mid (quot (+ lo hi) 2)
            v (nth arr mid)]
        (cond
          (= v target) mid
          (< v target) (recur (inc mid) hi)
          :else        (recur lo (dec mid)))))))

(let [arr [1 2 3 4 5 6 7 8 9 10]
      target 7
      idx (binary-search arr target)]
  (if idx
    (println (str "Found " target " at index " idx))
    (println (str target " not found"))))
